.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```

Comment:
What do these two steps do? Also I think they might have some warnings when they run looking at [the check log](https://github.com/bean-pod/switchboard/pull/87/checks?check_run_id=1229408306).

Reply:
These two steps allow caching dependencies and build outputs to improve workflow execution time. I did not see any warnings related to these two steps in [the check log](https://github.com/bean-pod/switchboard/pull/87/checks?check_run_id=1229408306). You can read on actions/cache [here](https://github.com/actions/cache) to get all the details. The warning I believe you are referring to is related to spring boot.